examples/solverdummies/c/solverdummy.c

```diff
@@ -6,50 +6,62 @@
 
 int main(int argc, char **argv)
 {
-  double  dt                 = 0.0;
-  int     solverProcessIndex = 0;
-  int     solverProcessSize  = 1;
-  int     dimensions         = -1;
-  double *vertices;
-  double *readData;
-  double *writeData;
-  int     meshID = -1;
-  int     dataID = -1;
-  int *   vertexIDs;
-  int     numberOfVertices = 3;
-  int     writeDataID      = -1;
-  int     readDataID       = -1;
-
-  if (argc != 4) {
+  double      dt                 = 0.0;
+  int         solverProcessIndex = 0;
+  int         solverProcessSize  = 1;
+  int         dimensions         = -1;
+  double *    vertices;
+  double *    readData;
+  double *    writeData;
+  int         meshID = -1;
+  int         dataID = -1;
+  int *       vertexIDs;
+  int         numberOfVertices = 3;
+  int         writeDataID      = -1;
+  int         readDataID       = -1;
+  const char *meshName;
+
+  const char *configFileName  = argv[1];
+  const char *participantName = argv[2];
+
+  if (argc != 3) {
     printf("Usage: ./solverdummy configFile solverName meshName\n\n");
     printf("Parameter description\n");
     printf("  configurationFile: Path and filename of preCICE configuration\n");
     printf("  solverName:        SolverDummy participant name in preCICE configuration\n");
-    printf("  meshName:          Mesh in preCICE configuration that carries read and write data\n");
     return 1;
   }
 
-  const char *configFileName  = argv[1];
-  const char *participantName = argv[2];
-  const char *meshName        = argv[3];
-
-  printf("DUMMY: Running solver dummy with preCICE config file \"%s\", participant name \"%s\", and mesh name \"%s\".\n",
-         configFileName, participantName, meshName);
+  printf("DUMMY: Running solver dummy with preCICE config file \"%s\" and participant name \"%s\".\n",
+         configFileName, participantName);
 
   const char *writeItCheckp = precicec_actionWriteIterationCheckpoint();
   const char *readItCheckp  = precicec_actionReadIterationCheckpoint();
 
   precicec_createSolverInterface(participantName, configFileName, solverProcessIndex, solverProcessSize);
 
+  if (strcmp(participantName, "SolverOne") == 0) {
+    meshName = "MeshOne";
+  }
+  if (strcmp(participantName, "SolverTwo") == 0) {
+    meshName = "MeshTwo";
+  }
```

Comment:
```suggestion
  if (participantName == "SolverOne") {
    meshName = "MeshOne";
  }
  if (participantName == "SolverTwo") {
    meshName = "MeshTwo";
  }
```
Looks more readable to me and should give the same result.
See https://stackoverflow.com/a/9159670